e2e/test/scenarios/dashboard-filters/reproductions/20049-change-question-parameter-type-affects-dashboard-filter.cy.spec.js

```diff
@@ -0,0 +1,125 @@
+import { produce } from "immer";
+import { restore, visitDashboard, visitQuestion } from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+
+const { PRODUCTS } = SAMPLE_DATABASE;
+
+const filterDetails = {
+  name: "Category",
+  slug: "category",
+  id: "c32a49e1",
+  type: "string/contains",
+  default: ["Doohickey"],
+};
+
+const questionDetails = {
+  name: "Question",
+  native: {
+    query: "select * from products where {{category}} limit 10",
+    "template-tags": {
+      category: {
+        id: "6b8b10ef-0104-1047-1e5v-2492d5954555",
+        name: "category",
+        "display-name": "Category",
+        type: "dimension",
+        dimension: ["field", PRODUCTS.CATEGORY, null],
+        "widget-type": "string/contains",
+      },
+    },
+  },
+};
+
+const dashboardDetails = {
+  name: "Dashboard #20049",
+  parameters: [filterDetails],
+};
+const dashcardDetails = {
+  row: 0,
+  col: 0,
+  size_x: 16,
+  size_y: 8,
+};
+
+describe.skip("issue 20049", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.createNativeQuestionAndDashboard({
+      questionDetails,
+      dashboardDetails,
+    }).then(({ body: { id, card_id, dashboard_id } }) => {
+      cy.wrap(dashboard_id).as("dashboardId");
+      cy.wrap(card_id).as("cardId");
+
+      cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+        cards: [
+          {
+            id,
+            card_id,
+            ...dashcardDetails,
+            parameter_mappings: [
+              {
+                card_id,
+                parameter_id: filterDetails.id,
+                target: ["dimension", ["template-tag", filterDetails.slug]],
+              },
+            ],
+          },
+        ],
+      });
+    });
+  });
+
+  it("Filter should stop applying if mapped question parameter is changed (metabase#20049)", () => {
+    cy.get("@cardId").then(cardId => {
+      updateQuestion(cardId);
+      visitQuestion(cardId);
+      verifyParameterType("String does not contain");
```

Comment:
I think this expectation is incorrect. Dashboard filters have their own type (see the screenshot)
![image](https://github.com/metabase/metabase/assets/1937582/af6ad4cc-dad9-40a8-b8c9-bfe8fd7c4e40)

Once it's added, its type never changes. A single dashboard filter can connect to multiple questions. To expect that the dashboard filter would change according to one of its mapped question feels wrong.

What should happen in my opinion is to drop the association in the UI and these are what could happen next.
1. Users map dashboard string contains filter to another question -> the `parameter_mappings` will be updated so that the association for the question with parameter widget type strings not contain will be gone.
2. Users modify the first question's parameter widget type back to string contains, the dashboard should act as if the association is still valid, and the dashboard filter string contains should show that it's still mapped to the question.

Can you ask product for clarification on this behavior? tl;dr I don't think we should ever expect the dashboard filter type to change.

Reply:
I needed to be clear, but as far as I understand your question, you suppose we're going to change dashboard filter type accordingly with question parameter type change. if so, it's not correct, we only want to "disconnect" dashboard filter and question parameter type if they do not match, so filter is not going to be applied to the card. Currently the filter is applied to the card even if parameter type of the question is changed and obviously filter is not correct

Reply:
Sorry for the confusion. I initially thought `verifyParameterType` was tested against the dashboard filter, I see now that it's tested against the question parameter. This repro looks correct. And you understand it correctly.

I think we're already on the same page, but I was mistaken.